cmd/mdatagen/internal/templates/metrics.go.tmpl

```diff
@@ -120,24 +123,45 @@ func (m *metric{{ $name.Render }}) init() {
 
 func (m *metric{{ $name.Render }}) recordDataPoint(start pcommon.Timestamp, ts pcommon.Timestamp, val {{ $metric.Data.MetricValueType.BasicType }}
 {{- range $metric.Attributes -}}{{- if not (attributeInfo .).Optional -}}, {{ .RenderUnexported }}AttributeValue {{ (attributeInfo .).Type.Primitive }}{{ end }}{{ end }}{{- if $metric.HasOptionalAttribute $.Attributes -}}, options ...MetricAttributeOption{{- end -}}) {
-	if !m.config.Enabled {
-		return
-	}
-	dp := m.data.{{ $metric.Data.Type }}().DataPoints().AppendEmpty()
+  if !m.config.Enabled {
+    return
+  }
+  nm := pmetric.NewMetric()
+	nm.SetName("{{ $name }}")
+	nm.SetDescription("{{ $metric.Description }}")
+	nm.SetUnit("{{ $metric.Unit }}")
+	nm.SetEmpty{{ $metric.Data.Type }}()
+	{{- if $metric.Data.HasMonotonic }}
+	nm.{{ $metric.Data.Type }}().SetIsMonotonic({{ $metric.Data.Monotonic }})
+	{{- end }}
+	{{- if $metric.Data.HasAggregated }}
+	nm.{{ $metric.Data.Type }}().SetAggregationTemporality(pmetric.AggregationTemporality{{ $metric.Data.AggregationTemporality }})
+	{{- end }}
+	{{- if $metric.Attributes }}
+	nm.{{ $metric.Data.Type }}().DataPoints().EnsureCapacity(m.capacity)
+	{{- end }}
+	dp := nm.{{ $metric.Data.Type }}().DataPoints().AppendEmpty()
```

Comment:
We can create it separately for now with `pmetric.NewNumberDataPoint()`, then either ignore or move to `m.data.{{ $metric.Data.Type }}().DataPoints().AppendEmpty()`